### PR TITLE
Adding FTP Put Action and unit test

### PR DESCRIPTION
**Pull request by gokulavasan**
Also, refactor some common FTP code and also PluginConfig properties.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,11 +7,28 @@
   <version>1.1.1</version>
   <name>ftp-copy-action</name>
   <url>http://maven.apache.org</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <docs.dir>docs</docs.dir>
+    <etl.versionRange>[3.3.0,10.0.0-SNAPSHOT)</etl.versionRange>
+    <app.parents>
+      system:cdap-etl-batch,
```
Comment (anwar6953) [inline pom.xml]: isn't etl-batch deprecated?

Comment (gokulavasan) [reply pom.xml]: Let me check with @sagarkapare, I just moved that part of the pom.xml.

File: src/main/java/co/cask/hydrator/action/ftp/FTPCopyAction.java [hunk 2]
```diff
@@ -58,79 +55,35 @@ public FTPCopyAction(FTPCopyActionConfig config) {
...
 
-    public String getDestDirectory() {
-      return destDirectory;
-    }
-
-    public int getPort() {
-      return (port != null) ? Integer.parseInt(port) : 21;
-    }
-
     public String getProtocol() {
```
Comment (sagarkapare) [inline src/main/java/co/cask/hydrator/action/ftp/FTPCopyAction.java]: not added in this PR.. but this method is not used anymore I think and can be removed.

File: widgets/FTPPut-action.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "metadata": {
+    "spec-version": "1.0"
+  },
+  "configuration-groups": [
+    {
+      "label": "FTP Copy Properties",
```
Comment (sagarkapare) [inline widgets/FTPPut-action.json]: FTP Put Properties

**Review by sagarkapare — APPROVED** [review]
LGTM 👍

**Comment by gokulavasan** [comment]
@sreevatsanraman Should I bump up the version of ftp-copy-action artifact?

**Pull request merged**